parsers/gdscript.c

```diff
@@ -0,0 +1,1343 @@
+
+	if (decorators && vStringLength (decorators) > 0)
+	{
+		attachParserField (e, false, GDScriptFields[F_ANNOTATIONS].ftype,
+						   vStringValue (decorators));
+	}
+
+	vString *vtype = vStringNew();
+	char * stype = vStringValue (type);
+	if (keyword == KEYWORD_const) {
+		if (strcmp(stype, "=") && strcmp(stype, "")) {
+			vStringCatS(vtype, "const ");
+			vStringCatS(vtype, stype);
+		} else
+			vStringCatS(vtype, "const");
+		vStringDeleteUnwrap (type);
+	} else {
+		if (strcmp(stype, "=") && strcmp(stype, ""))
+			vtype = type;
+	}
+
+
+	if (e && vStringLength(vtype) > 0) /// TODO: Fix types away
+	{
+		e->extensionFields.typeRef [0] = eStrdup ("typename");
+		e->extensionFields.typeRef [1] = vStringDeleteUnwrap (vtype);
+	}
+
+
+	while ((TokenContinuationDepth > 0 || token->type != ',') &&
+		   token->type != TOKEN_EOF &&
+		   token->type != ';' &&
+		   token->type != TOKEN_INDENT)
+	{
+		readToken (token);
+	}
+
+
+	return false;
+}
+
+/* pops any level >= to indent */
+static void setIndent (tokenInfo *const token)
+{
+	NestingLevel *lv = nestingLevelsGetCurrent (GDScriptNestingLevels);
+
+	while (lv && PY_NL (lv)->indentation >= token->indent)
+	{
+		tagEntryInfo *e = getEntryInCorkQueue (lv->corkIndex);
+		if (e)
+			e->extensionFields.endLine = token->lineNumber;
+
+		nestingLevelsPop (GDScriptNestingLevels);
+		lv = nestingLevelsGetCurrent (GDScriptNestingLevels);
+	}
+}
+
+static void findGDScriptTags (void)
+{
+	tokenInfo *const token = newToken ();
+	vString *decorators = vStringNew ();
+	bool atStatementStart = true;
+
+	TokenContinuationDepth = 0;
+	NextToken = NULL;
+	GDScriptNestingLevels = nestingLevelsNew (sizeof (struct gdscriptNestingLevelUserData));
+
+	readToken (token);
+	while (token->type != TOKEN_EOF)
+	{
+		tokenType iterationTokenType = token->type;
+		bool readNext = true;
+
+		if (token->type == TOKEN_INDENT)
+			setIndent (token);
+		else if (token->keyword == KEYWORD_class ||
+		         token->keyword == KEYWORD_func)
+		{
+			gdscriptKind kind = token->keyword == KEYWORD_class ? K_CLASS : K_FUNCTION;
+
+			readNext = parseClassOrDef (token, decorators, kind, false);
+		}
+		else if (token->keyword == KEYWORD_extends)
+		{
+			readNext = parseExtends (token);
+		}
+		else if (token->type == '(')
+		{ /* skip parentheses to avoid finding stuff inside them */
+			readNext = skipOverPair (token, '(', ')', NULL, false);
+		}
+		else if (token->keyword == KEYWORD_variable || token->keyword == KEYWORD_const)
+		{
+			NestingLevel *lv = nestingLevelsGetCurrent (GDScriptNestingLevels);
+			tagEntryInfo *lvEntry = getEntryOfNestingLevel (lv);
+			gdscriptKind kind = K_VARIABLE;
+
+			if (lvEntry && lvEntry->kindIndex != K_CLASS)
+				kind = K_LOCAL_VARIABLE;
+
+			readNext = parseVariable (token, kind, decorators, token->keyword);
+		}
+		else if (token->keyword == KEYWORD_enum)
+		{
+			readNext = parseEnum (token);
+		}
+		else if (token->type == '@' && atStatementStart &&
```

Comment:
https://godotengine.org/article/gdscript-progress-report-new-gdscript-now-merged
It seems that GDScript has the "annotation" feature. Could you add a test case for testing "@..."?
So the test coverage increases.

Reply:
Oh whoops, I had some annotations in the first test I was using, slipped the mind when I grabbed the docs sample. I can add some annotations sure

Reply:
I confirmed that the annotations are attached to tags in the test cases.